project_tracker_template/update_gantt.gs

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Updates all the custom formulas in the Timeline sheet
+ * @param {Object[][]} ganttTableValues A two dimensional array containing
+ *     information about tasks and milestones
+ * @param {Number} numRows Number of rows containing milestone/ task data
+ */
+function updateGantt(ganttTableValues, numRows) {
+  for (var i = 0; i < ganttTableValues.length; i++) {
+    for (var j = 0; j < 13; j++) {
+      ganttTableValues[i][j] = [ '' ];
+    }
+  }
+  // Set Project Title
+  ganttTableValues[0][1] = [ '=Tasks!B1' ];
+  // Set Project Start Date
+  ganttTableValues[0][2] = [ '=Tasks!H2' ];
+  // Set Estimated Project Launch Date
+  ganttTableValues[0][3] = [ '=Tasks!H4' ];
+  // Set custom formula for week wise representation of the project timeline
+  ganttTableValues[0][4] = [
+    '=iferror(SPARKLINE({split(rept("7,",FLOOR((int(D6)-int(C6))/7)),","),mod(int(D6)-int(C6),7)},{"charttype","bar";"color1","white";"color2","lightblue"}),"")'
+  ];
+  ganttTableValues[0][8] = [ 'Milestone' ];
+  // Set Task Serial Number from Task Sheet
+  ganttTableValues[1][0] = [ '=ArrayFormula(Tasks!$A$7:$A)' ];
+  // Set Task/ Milestone Title from Task Sheet
+  ganttTableValues[1][1] = [ '=ArrayFormula(Tasks!$B$7:$B)' ];
+  // Set Task/ Milestone Start Date from Task Sheet
+  ganttTableValues[1][2] = [ '=ArrayFormula(Tasks!$F$7:$F)' ];
+  // Set Task/ Milestone Estimated Launch Date from Task Sheet
+  ganttTableValues[1][3] = [ '=ArrayFormula(Tasks!$G$7:$G)' ];
+  // Set Task/ Milestone Status from Task Sheet
+  ganttTableValues[1][5] = [ '=ArrayFormula(Tasks!$H$7:$H)' ];
+  // Set Task/ Milestone Owner from Task Sheet
+  ganttTableValues[1][6] = [ '=ArrayFormula(Tasks!$C$7:$C)' ];
+  // Set the usernames of team members in the Color Legend Block
+  ganttTableValues[1][8] = [ '=ArrayFormula(Team!$C$6:$C$25)' ];
```

Comment:
might be a good idea to create a named range for C6:C25 and use it. similarly for the ranges in the above formula. This will then take care of any newly added columns / removed column etc